gradle-versions/src/main/java/com/markelliot/gradle/versions/RootUpdateVersionsPlugin.java

```diff
@@ -0,0 +1,78 @@
+package com.markelliot.gradle.versions;
+
+import java.util.Map;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+
+public final class RootUpdateVersionsPlugin implements Plugin<Project> {
+    private static final String VERSIONS_PROPS = "versions.props";
+    private static final String ALL_REPORTS = "allReports";
+
+    @Override
+    public void apply(Project project) {
+        if (!project.equals(project.getRootProject())) {
+            throw new IllegalStateException("Plugin must be applied onto root project");
```

Comment:
Probably have to make this a no-op + log line instead since the stuff in the wild does things like https://github.com/markelliot/barista/blob/develop/build.gradle.kts#L32